blockstore.go

```diff
@@ -84,32 +87,31 @@ func (b *Blockstore) Has(cid cid.Cid) (bool, error) {
 	return ret, err
 }
 
-func (b *Blockstore) Get(cid cid.Cid) (blocks.Block, error) {
+func (b *Blockstore) Get(ctx context.Context, cid cid.Cid) (blocks.Block, error) {
 	var data []byte
 	switch err := b.prepared[stmtGet].QueryRow(keyFromCid(cid)).Scan(&data); err {
 	case sql.ErrNoRows:
-		return nil, blockstore.ErrNotFound
+		return nil, ErrNotFound
```

Comment:
Please return `ipld.ErrNotFound{Cid: k}`. ErrNotFound errors are bubbled and layers above rely on checking whether is ipld.ErrNotFound.

Reply:
Ref: https://github.com/ipfs/boxo/blob/v0.27.4/blockstore/blockstore.go#L182-L206

Reply:
No because this is a blockstore layer, not a datastore layer..